core/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/runtimeModels.scala

```diff
@@ -585,3 +585,34 @@ object RuntimeOperation {
     val asString = "stopping"
   }
 }
+
+final case class DataProcMachineSpecification(machineTypeName: MachineTypeName, cpu: Int, memory: Double)
+
+object MachineTypes {
```

Comment:
Just want to point to some [existing code](https://github.com/DataBiosphere/leonardo/blob/c8f1292698db194169e69c0465670b6fde1355c1/http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GceInterpreter.scala#L515) which dynamically resolves the machine type in GCE to get the CPU/memory… could be a way to avoid enumerating them all.

Reply:
Seems like this is a reasonable approach! Seems to allocate more memory for the system / welder than I was planning to. 
I also switched to megs over gigs for better precision.

Reply:
Hey @rtitle , I'm a bit confused about the output of the getClusterResourceConstraints. 
I'm not sure how the total - dataprocallocated - welder allocated is being used.

Essentially what I want to do is set `spark:spark.driver.memory` to the most amount of memory possible for the machine. And from what I understand, this is what "dataproc" is.

So with getClusterResourceConstraints, it seems like this 6gb is being subtracted from available memory to reserve for dataproc... but when I go to use this function to get data for dataproc, I'm actually reserving memory for dataproc (The spark JVM).
For lower memory machines, this 6gb subtraction is killer. For higher memory machines, it's less significant, and results from my previous implementation will be closer to this implementation.

What am I missing?